include/zephyr/net/net_if.h

```diff
@@ -2643,6 +2643,35 @@ void net_if_unregister_link_cb(struct net_if_link_cb *link);
 void net_if_call_link_cb(struct net_if *iface, struct net_linkaddr *lladdr,
 			 int status);
 
+/* used to ensure encoding of checksum support in net_if.h and
+ * ethernet.h is the same
+ */
+#define NET_IF_CHECKSUM_NONE_BIT			0
+#define NET_IF_CHECKSUM_IPV4_HEADER_BIT			BIT(0)
+#define NET_IF_CHECKSUM_IPV4_ICMP_BIT			BIT(1)
+#define NET_IF_CHECKSUM_IPV6_HEADER_BIT			BIT(10)
+#define NET_IF_CHECKSUM_IPV6_ICMP_BIT			BIT(11)
+#define NET_IF_CHECKSUM_TCP_BIT				BIT(21)
+#define NET_IF_CHECKSUM_UDP_BIT				BIT(22)
+
+/**
+ * @brief Type of checksum for which support in the interface will be queried.
+ */
+enum net_if_checksum_type {
+	NET_IF_CHECKSUM_IPV4_HEADER = NET_IF_CHECKSUM_IPV4_HEADER_BIT,
+	NET_IF_CHECKSUM_IPV4_TCP = NET_IF_CHECKSUM_IPV4_HEADER_BIT|
+		NET_IF_CHECKSUM_TCP_BIT,
+	NET_IF_CHECKSUM_IPV4_UDP = NET_IF_CHECKSUM_IPV4_HEADER_BIT|
+		NET_IF_CHECKSUM_UDP_BIT,
```

Comment:
```suggestion
	NET_IF_CHECKSUM_IPV4_UDP = NET_IF_CHECKSUM_IPV4_HEADER_BIT |
				   NET_IF_CHECKSUM_UDP_BIT,
```
etc, just reads better.

Reply:
Done